### PR TITLE
GSoC'21 Quarter Progress: Aitik Gupta

### PR DESCRIPTION
**Pull request by aitikgupta**
This time I went with a custom meme. :p
image

### CODE REVIEW

File: content/posts/GSoC_2021_Quarter/index.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+> Assumption: the Chinese font doesn't have those English glyphs, and vice versa
+
+With this short writeup, I'll talk about how does a migration from a font-first to a text-first approach in Matplotlib looks like, which ideally solves the above problem.
+### Have the fonts?
+Logically, the very first step to solving this would be to ask whether you _have_ multiple fonts, right?
+
+Matplotlib doesn't ship [CJK](https://en.wikipedia.org/wiki/List_of_CJK_fonts) (Chinese Japanese Korean) fonts, which ideally contains these Chinese glyphs. It does try to cover most grounds with the [default font](https://matplotlib.org/stable/users/dflt_style_changes.html#normal-text) it ships with, however.
+
+So if you don't have a font to render your Chinese characters, go ahead and install one! Matplotlib will find your installed fonts.
```
Comment (story645) [inline content/posts/GSoC_2021_Quarter/index.md]: Is this true out of the box or do you need to rebuild the font cache?

File: content/posts/GSoC_2021_Quarter/index.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    <figure>
+        <img src="https://user-images.githubusercontent.com/43996118/128347495-1f4f858d-33d3-4119-8732-5b26c4e9ca2a.png" alt="ChineseInBetween" />
+        <figcaption style="text-align: center; font-style: italic;">PNG straight outta Matplotlib!</figcaption>
+    </figure>
+</p>
+
+## Python C-API is hard, at first!
+I've spent days at Python C-API's [argument doc](https://docs.python.org/3/c-api/arg.html), and it's hard to get what you need at first, ngl.
+
+But, with the help of some amazing people in GSoC community ([@srijan-paul](https://srijan-paul.github.io/), [@atharvaraykar](https://atharvaraykar.me/)) and amazing mentors, blockers begone!
```
Comment (story645) [inline content/posts/GSoC_2021_Quarter/index.md]:
```suggestion
But, with the help of some amazing people in the GSoC community ([@srijan-paul](https://srijan-paul.github.io/), [@atharvaraykar](https://atharvaraykar.me/)) and amazing mentors, blockers begone!
```

also 😁 that's so awesome!

File: content/posts/GSoC_2021_Quarter/index.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+resources:
+- name: featuredImage
+  src: "AitikGupta_GSoC.png"
+  params:
+    showOnTop: true
+---
+
+**“<ins>Matplotlib, I want 多个汉字 in between my text.</ins>”**
+
+Let's say you wanted some (多个汉字 -> multiple Chinese characters) in between your English text, to be rendered by Matplotlib.
```
Comment (story645) [inline content/posts/GSoC_2021_Quarter/index.md]:
```suggestion
Let's say you wanted Matplotlib to render 多个汉字 (multiple Chinese characters) in between your English text.
```
I think might be cleaner to go your standard example, something like  
Let's say you wanted to label something on a plot "there is 多个汉字 in between", which is multiple Chinese characters in between English  characters.

File: content/posts/GSoC_2021_Quarter/index.md (hunk 2 above)
Comment (story645) [reply content/posts/GSoC_2021_Quarter/index.md]: also to remind myself for when this gets tweeted out, @_injuly & @AtharvaRaykar

File: content/posts/GSoC_2021_Quarter/index.md (hunk 1 above)
Comment (aitikgupta) [reply content/posts/GSoC_2021_Quarter/index.md]: We do need to rebuild the cache, but the statement was meant to be a generic one.. that Matplotlib _can_ find your custom installed fonts.

File: content/posts/GSoC_2021_Quarter/index.md (hunk 2 above)
Comment (aitikgupta) [reply content/posts/GSoC_2021_Quarter/index.md]: 😄 Cool

File: content/posts/GSoC_2021_Quarter/index.md (hunk 3 above)
Comment (aitikgupta) [reply content/posts/GSoC_2021_Quarter/index.md]: Let me fix that

File: content/posts/GSoC_2021_Quarter/index.md (hunk 1 above)
Comment (aitikgupta) [reply content/posts/GSoC_2021_Quarter/index.md]: Modified:
```txt
Matplotlib will find your installed fonts (after rebuilding the cache, that is).
```

**Review by story645 — APPROVED** [review]

**Comment by aitikgupta** [comment]
Should we merge it? @davidevaleriani

**Comment by davidevaleriani** [comment]
I'm currently on vacation until Aug 17.

Davide

**Comment by aitikgupta** [comment]
Oh, happy vacations @davidevaleriani!

(I need the URL to put in NumFOCUS' spreadsheet though, @story645 could you merge it?)

**Comment by story645** [comment]
Sorry about that @davidevaleriani ,
Here you go @aitikgupta https://matplotlib.org/matplotblog/posts/gsoc_2021_quarter/

**Pull request merged**